src/main/drivers/flash.c

```diff
@@ -208,39 +335,65 @@ static bool flashSpiInit(const flashConfig_t *flashConfig)
 
     return false;
 }
-#endif // USE_SPI
+#endif // USE_FLASH_SPI
+
+void flashPreInit(const flashConfig_t *flashConfig)
+{
+    spiPreinitRegister(flashConfig->csTag, IOCFG_IPU, 1);
+}
 
 bool flashDeviceInit(const flashConfig_t *flashConfig)
 {
-#ifdef USE_SPI
+    bool haveFlash = false;
+
+#ifdef USE_FLASH_SPI
     bool useSpi = (SPI_CFG_TO_DEV(flashConfig->spiDevice) != SPIINVALID);
 
     if (useSpi) {
-        return flashSpiInit(flashConfig);
+        haveFlash = flashSpiInit(flashConfig);
     }
 #endif
 
-#ifdef USE_QUADSPI
+#ifdef USE_FLASH_QUADSPI
     bool useQuadSpi = (QUADSPI_CFG_TO_DEV(flashConfig->quadSpiDevice) != QUADSPIINVALID);
     if (useQuadSpi) {
```

Comment:
Old code did return on first flash found
```suggestion
    if (!haveFlash && useQuadSpi) {
```
Maybe use `do { .. } while(0);` and break on first `haveFlash`